### PR TITLE
Some improvements about names of span logs

### PR DESCRIPTION
**Pull request by FrankChen021**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Improvement name of some span logs

#### 1. use type of job name as span name

Thread name is used as name of the span that is first created on a thread. If a Job does not set the thread name by calling `setThreadName` inside it, the default name 'ThreadPool::worker' is used. 
We often see a lot of logs with this default name, and such default name provides nothing to understand the execution flow.

image

So, there's an improvement on this part that if the thread name is not changed by the job, the type name of the job will be used as span log. This gives a very clear information that what the thread is doing.

#### 2. use name of IProcessors only as span name

Previously, `ExecutionThreadContext::executeTask()` is added as prefix of the span name, this makes the log not so intuitive.
This prefix is eliminated and only the name of processor itself is used.

#### 3. add a span for PipelineExecutor.execute to give a clear info that how many threads are forked during the execution of pipeline.

These improvements are illustrated on the pic below

image

Since these names are related to the class name, and they're might change over time, it does not make any sense to write test cases to cover the changes in this PR.

### CODE REVIEW

File: src/Common/ThreadPool.cpp [hunk 1]
```diff
@@ -271,7 +273,7 @@ void ThreadPoolImpl<Thread>::worker(typename std::list<Thread>::iterator thread_
     while (true)
     {
         /// This is inside the loop to also reset previous thread names set inside the jobs.
-        setThreadName("ThreadPool");
+        setThreadName(DEFAULT_THRAED_NAME);
```
Comment (qoega) [inline src/Common/ThreadPool.cpp:276]:
```suggestion
        setThreadName(DEFAULT_THREAD_NAME);
```

File: src/Common/ThreadPool.cpp [hunk 2]
```diff
@@ -317,15 +319,23 @@ void ThreadPoolImpl<Thread>::worker(typename std::list<Thread>::iterator thread_
                 if (thread_trace_context.root_span.isTraceEnabled())
                 {
                     /// Use the thread name as operation name so that the tracing log will be more clear.
-                    /// The thread name is usually set in the jobs, we can only get the name after the job finishes
+                    /// The thread name is usually set in jobs, we can only get the name after the job finishes
                     std::string thread_name = getThreadName();
-                    if (!thread_name.empty())
+                    if (!thread_name.empty() && thread_name != DEFAULT_THRAED_NAME)
```
Comment (qoega) [inline src/Common/ThreadPool.cpp:324]:
```suggestion
                    if (!thread_name.empty() && thread_name != DEFAULT_THREAD_NAME)
```

File: src/Processors/Executors/ExecutionThreadContext.cpp [hunk 3]
```diff
@@ -75,7 +75,7 @@ bool ExecutionThreadContext::executeTask()
 
     if (trace_processors)
     {
-        span = std::make_unique<OpenTelemetry::SpanHolder>("ExecutionThreadContext::executeTask() " + node->processor->getName());
+        span = std::make_unique<OpenTelemetry::SpanHolder>(node->processor->getName());
```
Comment (qoega) [inline src/Processors/Executors/ExecutionThreadContext.cpp:78]: Maybe leave some small prefix like "P:" for processors?

File: src/Common/ThreadPool.cpp (hunk 1 above)
Comment (FrankChen021) [reply src/Common/ThreadPool.cpp:276]: Good catch.

File: src/Processors/Executors/ExecutionThreadContext.cpp (hunk 3 above)
Comment (FrankChen021) [reply src/Processors/Executors/ExecutionThreadContext.cpp:78]: The names in span in most cases are function names. So 'P:' seems a little bit wired for users to understand. I think using the processor name itself is clearly. 

If we really want to give a clear information that this span is processor, how about using the function name style by adding prefix 'Processor::' ?

**Comment by FrankChen021** [comment]
I checked log of the failed test, seems that it has nothing to do with the changes in the PR

https://s3.amazonaws.com/clickhouse-test-reports/47667/87b548cb71fb5461521a9f108f46017d327a0ab4/integration_tests__tsan__[1/6].html

**Comment by nickitat** [comment]
Stateless tests (debug) [4/5] - `../contrib/s2geometry/src/s2/s2latlng_rect.h:382 ERROR Invalid rect: [50.4210420, 0.4931979], [2.1000000, 2.9000000]
Stateless tests (debug, s3 storage) [5/6] - timeout, occured few times in the last days 
../contrib/s2geometry/src/s2/s2latlng.cc:38 ERROR Invalid S2LatLng in S2LatLng::ToPoint: [1.84467e+19, 1.84467e+19]`
Stateless tests (tsan, s3 storage) [1/5] - should be fixed by #47953

**Review by nickitat — APPROVED** [review]

**Comment by FrankChen021** [comment]
Thank you @nickitat

**Pull request merged**